engine/sdk_tx_getter.go

```diff
@@ -0,0 +1,51 @@
+package engine
+
+import (
+	"context"
+	"slices"
+	"time"
+
+	sdk "github.com/bitcoin-sv/go-sdk/transaction"
+	"github.com/bitcoin-sv/spv-wallet/engine/spverrors"
+	"iter"
+)
+
+type sdkTxGetter struct {
+	client *Client
+}
+
+func newSDKTxGetter(client *Client) *sdkTxGetter {
+	return &sdkTxGetter{client: client}
+}
+
+func (g *sdkTxGetter) GetTransactions(ctx context.Context, ids iter.Seq[string]) ([]*sdk.Transaction, error) {
+	db := g.client.Datastore().DB()
+
+	queryIDsCtx, cancel := context.WithTimeout(ctx, 20*time.Second)
+	defer cancel()
+
+	var hexes []struct {
+		Hex string
+	}
+
+	err := db.
+		WithContext(queryIDsCtx).
+		Model(&Transaction{}).
+		Where("id IN (?)", slices.Collect(ids)).
+		Find(&hexes).
+		Error
+
+	if err != nil {
+		return nil, spverrors.Wrapf(err, "Cannot get transactions by IDs from database")
+	}
+
+	transactions := make([]*sdk.Transaction, 0, len(hexes))
+	for _, record := range hexes {
+		tx, err := sdk.NewTransactionFromHex(record.Hex)
+		if err != nil {
+			return nil, spverrors.Wrapf(err, "Cannot parse transaction hex")
+		}
+		transactions = append(transactions, tx)
+	}
```

Comment:
```suggestion
	transactions := make([]*sdk.Transaction, 0, len(hexes))
	for _, record := range hexes {
		tx, err := sdk.NewTransactionFromHex(record.Hex)
		if err != nil {
			return nil, spverrors.Wrapf(err, "Cannot parse transaction hex")
		}
		transactions = append(transactions, tx)
	}
```
```go
transactions := make([]*sdk.Transaction, 0, len(hexes))
mu := sync.Mutex{}
wg := sync.WaitGroup{}

for _, record := range hexes {
    wg.Add(1)
    go func(hex string) {
        defer wg.Done()
        tx, err := sdk.NewTransactionFromHex(hex)
        if err != nil {
            // Handle error (could log or collect the error for later return)
            return
        }
        mu.Lock()
        transactions = append(transactions, tx)
        mu.Unlock()
    }(record.Hex)
}

wg.Wait()
```
something to consider

Reply:
Hmmm 🤔 
Regarding the size of `hexes` - which usually is no more than several elements, I would not introduce it 🤞 

So, for now, it seems like premature optimization which (considering (light but non-zero) overhead of goroutines and another sync mechanizm) might work even slower.

I suppose, in the future, we will have to review the whole transaction-flow (after current remaking) and try to boost/optimize things up.  However, I would start with locating bottlenecks and then optimizing.

Reply:
ok